### PR TITLE
feat: web3 profile setting UI

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-1838

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/UI/components/index.ts [hunk 1]
```diff
@@ -29,3 +29,4 @@ export * from './WalletSettingCard/index.js'
 export * from './PluginCardFrame/index.js'
 export * from './Markdown/index.js'
 export * from './Icon/index.js'
+export * from './PublicWalletSetting'
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/index.ts]:
```suggestion
export * from './PublicWalletSettings.js'
```

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+        const [addresses, setAddresses] = useState<string[]>([])
+        const [showAlert, setShowAlert] = useState(true)
+        const { showSnackbar } = useCustomSnackbar()
+
+        const { value: hiddenAddresses, loading } = useHiddenAddressSetting(pluginId, currentPersona?.publicKeyAsHex)
+
+        const onSwitchChange = useCallback((address: string) => {
+            setAddresses((prev) => {
+                return prev.some((x) => address === x) ? prev.filter((x) => address !== x) : [...prev, address]
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/PublicWalletSetting/index.tsx]: Don't compare addresses with the equal sign.

File: packages/mask/src/components/InjectedComponents/ProfileTab/PublicWalletSettingEntry.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    button: {
+        display: 'inline-flex',
+        gap: theme.spacing(1),
+        minWidth: 254,
+        borderRadius: '20px',
+        width: 'fit-content !important',
+    },
+}))
+
+export const PublicWalletSettingEntry = memo(() => {
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/ProfileTab/PublicWalletSettingEntry.tsx]:
```suggestion
export const WalletSettingsEntry = memo(() => {
```

File: packages/plugin-infra/src/types.ts [hunk 4]
```diff
@@ -746,7 +746,7 @@ export namespace Plugin.SNSAdaptor {
     }
 
     export interface SettingTab {
```
Comment (guanbinrui) [inline packages/plugin-infra/src/types.ts:753]:
```suggestion
    export interface SettingsTab {
```

File: packages/plugins/Web3Profile/src/SNSAdaptor/index.tsx [hunk 5]
```diff
@@ -72,6 +72,9 @@ const sns: Plugin.SNSAdaptor.Definition = {
             ID: PluginId.Web3Profile,
             label: 'Web3Profile',
             priority: 2,
+            UI: {
+                TabContent: PublicWalletSetting,
```
Comment (guanbinrui) [inline packages/plugins/Web3Profile/src/SNSAdaptor/index.tsx:76]: The same `TabContent` has been exported in the `Tips` plugin.

File: packages/plugin-infra/src/types.ts (hunk 4 above)
Comment (Lanttcat) [reply packages/plugin-infra/src/types.ts:753]:
```
/** This UI will be rendered as tab on the setting dialog */
        SettingTabs?: SettingTab[]
```

I think should keep the `SettingTab`.

File: packages/plugins/Web3Profile/src/SNSAdaptor/index.tsx (hunk 5 above)
Comment (Lanttcat) [reply packages/plugins/Web3Profile/src/SNSAdaptor/index.tsx:76]: Yep, I also added in `Tips ` with `PublicWalletSetting `.

File: packages/mask/shared-ui/locales/en-US.json [hunk 6]
```diff
@@ -1205,6 +1205,10 @@
     "plugin_valuables_name": "Valuables",
     "plugin_valuables_description": "Buy & sell tweets autographed by their original creators.",
     "popups_wallet_unsupported_network": "Unsupported network type",
+    "plugin_default_title": "Default",
+    "plugin_provider_by": "Provided by",
     "plugin_enables": "Enable plugins",
+    "show_wallet_setting_intro": "Please select the wallet to active after saving.",
+    "show_wallet_setting_button": "Go Settings",
```
Comment (guanbinrui) [inline packages/mask/shared-ui/locales/en-US.json]: Go to Settings.

File: packages/shared/src/UI/components/index.ts [hunk 7]
```diff
@@ -35,3 +35,4 @@ export * from './FootnoteMenu/index.js'
 export * from './MaskPluginWrapper/index.js'
 export * from './MaskIcon/index.js'
 export * from './SourceProviderIcon/index.js'
+export * from './PublicWalletSetting/index.js'
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/index.ts:38]:
```suggestion
export * from './PublicWalletSettings/index.js'
```

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+interface PublicWalletSettingProps {
+    onClose: () => void
+    onOpenPopup: (route?: PopupRoutes, params?: Record<string, any>) => void
+    bindingWallets?: BindingProof[]
+    currentPersona?: ECKeyIdentifier
+    pluginId: MaskPluginID
+}
+
+export const PublicWalletSetting = memo<PublicWalletSettingProps>(
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/PublicWalletSetting/index.tsx:83]:
```suggestion
export const PublicWalletSettings = memo<PublicWalletSettingProps>(
```

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+    },
+    placeholderIcon: {
+        color: theme.palette.maskColor.third,
+    },
+    content: {
+        marginTop: 12,
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: 12,
+        '& > *': {
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/PublicWalletSetting/index.tsx]: Avoid using `*` if possible.

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+import {
+    BindingProof,
+    CrossIsolationMessages,
+    ECKeyIdentifier,
+    EMPTY_LIST,
+    NextIDPlatform,
+    PopupRoutes,
+} from '@masknet/shared-base'
+import { useHiddenAddressSetting, useWeb3State } from '@masknet/plugin-infra/web3'
+import { PluginID as MaskPluginID } from '@masknet/plugin-infra'
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/PublicWalletSetting/index.tsx]:
```suggestion
import { PluginID } from '@masknet/plugin-infra'
```

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+                const storage = Storage.createNextIDStorage(
+                    currentPersona.publicKeyAsHex,
+                    NextIDPlatform.NextID,
+                    currentPersona,
+                )
+                const prevResult = storage.get<PublicWalletSettingType>(MaskPluginID.Web3Profile)
+
+                await storage.set<PublicWalletSettingType>(MaskPluginID.Web3Profile, {
+                    ...prevResult,
+                    hiddenAddresses: addresses,
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/PublicWalletSetting/index.tsx]: Does it possible contains any dup addresses?

```suggestion
                    hiddenAddresses: uniqBy(addresses),
```

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx (hunk 8 above)
Comment (Lanttcat) [reply packages/shared/src/UI/components/PublicWalletSetting/index.tsx:83]: I think should remove 's', this component is only work for someone plugin.

File: packages/mask/src/components/InjectedComponents/PluginSettingsDialog.tsx [hunk 12]
```diff
@@ -82,19 +85,27 @@ export function PluginSettingsDialog() {
 
     useEffect(() => MaskMessages.events.ownPersonaChanged.on(retry), [retry])
 
+    const onOpenPopup = useCallback(Services.Helper.openPopupWindow, [])
```
Comment (UncleBill) [inline packages/mask/src/components/InjectedComponents/PluginSettingsDialog.tsx]: It's stable, no need to wrap with `useCallback`.

If you want an alias.
```suggestion
    const onOpenPopup = Services.Helper.openPopupWindow
```

File: packages/shared/src/UI/components/PublicWalletSetting/index.tsx [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+        const onOpenConnectWallet = useCallback(() => {
+            onOpenPopup(PopupRoutes.ConnectedWallets, {
+                internal: true,
+            })
+        }, [onOpenPopup])
+
+        const EmptyHintMapping = useMemo(() => {
+            const mapping: Record<string, string> = {
+                [PluginID.Tips]: t.add_wallet_tips(),
+                [PluginID.Tips]: t.add_wallet_web3_profile(),
```
Comment (UncleBill) [inline packages/shared/src/UI/components/PublicWalletSetting/index.tsx]:
```suggestion
                [PluginID.Tips]: t.add_wallet_tips(),
                [PluginID.Web3Profile]: t.add_wallet_web3_profile(),
```

File: packages/plugins/Web3Profile/src/SNSAdaptor/components/WalletSetting.tsx [hunk 14]
```diff
@@ -116,6 +116,7 @@ interface WalletSettingProp {
     retryData: () => void
 }
 
+// TODO: should remove this component
```
Comment (UncleBill) [inline packages/plugins/Web3Profile/src/SNSAdaptor/components/WalletSetting.tsx]: Will this TODO be left to be resolved in another PR?

**Review by guanbinrui — APPROVED** [review]
